_posts/2020-10-12-sparklyr-flint-0.2.0-released/sparklyr-flint-0.2.0.Rmd

```diff
@@ -0,0 +1,143 @@
+
+So, to summarize, the OLS regression functionality implemented in Flint not only outputs the solution to the problem, but also calculates useful metrics that helps data scentists assess the sanity and predictive quality of the resulting model.
+
+To see OLS regression in action in `sparklyr.flint`, one can run the following example:
+
+```
+mtcars_sdf <- copy_to(sc, mtcars, overwrite = TRUE) %>%
+  dplyr::mutate(time = 0L)
+mtcars_ts <- from_sdf(mtcars_sdf, is_sorted = TRUE, time_unit = "SECONDS")
+model <- ols_regression(mtcars_ts, mpg ~ hp + wt) %>% to_sdf()
+
+print(model %>% dplyr::pull(akaikeIC))
+## [1] 154.6523
+
+print(model %>% dplyr::pull(bayesIC))
+## [1] 159.0495
+
+print(model %>% dplyr::pull(cond))
+## [1] 345403.4
+
+## # ^ output says condition number of the Gram matrix was within reason
+
+print(model %>% dplyr::pull(beta))
+
+## [[1]]
+## [1] -0.03177295 -3.87783074
+```
+
+## Additional Summarizers
+
+The EWMA (Exponential Weighted Moving Average), EMA half-life, and the standardized moment summarizers (namely, skewness and kurtosis) along with a few others which were missing in `sparklyr.flint` 0.1 are now fully supported in `sparklyr.flint` 0.2.
+
+## Better Integration With `sparklyr`
+
+While `sparklyr.flint` 0.1 included a `collect()` method for exporting data from a Flint timeseries RDD to an R dataframe, it did not have a similar method that extracts the underlying Spark dataframe from a Flint timeseries RDD. This was clearly an oversight. In `sparklyr.flint` 0.2, one can call `to_sdf()` on a timeseries RDD to get back a Spark dataframe that is usable in `sparklyr` (e.g., as shown by `model %>% to_sdf() %>% dplyr::pull(...)` examples from above). One can also get to the underlying Spark dataframe JVM object reference by calling `spark_dataframe()` on a Flint timeseries RDD (this is usually unnecessary in vast majority of `sparklyr` use cases though).
```

Comment:
```suggestion
While `sparklyr.flint` 0.1 included a `collect()` method for exporting data from a Flint time-series RDD to an R data frame, it did not have a similar method for extracting the underlying Spark data frame from a Flint time-series RDD. This was clearly an oversight. In `sparklyr.flint` 0.2, one can call `to_sdf()` on a timeseries RDD to get back a Spark data frame that is usable in `sparklyr` (e.g., as shown by `model %>% to_sdf() %>% dplyr::pull(...)` examples from above). One can also get to the underlying Spark data frame JVM object reference by calling `spark_dataframe()` on a Flint time-series RDD (this is usually unnecessary in vast majority of `sparklyr` use cases though).
```

Reply:
I'm hyphenating time series because it's modifying RDD (which I assume is a noun).